extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/proxies/ProxyDefinitions.java

```diff
@@ -106,8 +108,45 @@ private static Class generateProxyClass(PersistentClass persistentClass, ByteBud
             return null;
         }
         final Set<Class> proxyInterfaces = ProxyFactoryHelper.extractProxyInterfaces(persistentClass, entityName);
-        Class proxyDef = byteBuddyProxyHelper.buildProxy(mappedClass, toArray(proxyInterfaces));
-        return proxyDef;
+        PreGeneratedProxies.ProxyClassDetailsHolder preProxy = preGeneratedProxies.getProxies()
+                .get(persistentClass.getClassName());
+        Class<?> preGeneratedProxy = null;
+        boolean match = true;
+        if (preProxy != null) {
+            match = proxyInterfaces.size() == preProxy.getProxyInterfaces().size();
+            if (match) {
+                for (Class i : proxyInterfaces) {
+                    if (!preProxy.getProxyInterfaces().contains(i.getName())) {
+                        match = false;
+                        break;
+                    }
+                }
+            }
+            if (match) {
+                try {
+                    preGeneratedProxy = Class.forName(preProxy.getClassName(), false,
+                            Thread.currentThread().getContextClassLoader());
+                } catch (ClassNotFoundException e) {
+                    //should never happen
+                    throw new RuntimeException("Unable to load proxy class", e);
+                }
+            }
+        }
+
+        if (preGeneratedProxy == null) {
+            if (match) {
+                LOGGER.warnf("Unable to find a build time generated proxy for entity %s",
```

Comment:
meaning add that to the message.

Reply:
+1

Also could we include the details of the mismatch, if any. I think we're assuming that this will never actually happen, but if it does it would be super useful to know the class which was expected, if we had that class at all registered in any `PreGeneratedProxies` , and if we had it what's the interfaces mismatch.

Reply:
I thought this could happen with Envars? This particular WARN would only happen if an entity was registered we don't know about.